dbt_copilot_helper/utils/validation.py

```diff
@@ -17,6 +18,62 @@ def validator(string):
     return validator
 
 
+def validate_s3_bucket_name(name):
+    return False
+
+
+def validate_addons(addons: dict):
+    """
+    Validate the addons file and return a dictionary of addon: error message.
+    """
+    errors = {}
+
+    for addon_name, addon in addons.items():
+        try:
+            addon_type = addon.get("type", None)
+            if not addon_type:
+                errors[addon_name] = f"Missing addon type in addon '{addon_name}'"
+                continue
+            schema = SCHEMA_MAP.get(addon_type, None)
+            if not schema:
+                errors[
+                    addon_name
+                ] = f"Unsupported addon type '{addon_type}' in addon '{addon_name}'"
+                continue
+            schema.validate(addon)
+        except SchemaError as ex:
+            errors[addon_name] = f"Error in {addon_name}: {ex.code}"
+
+    return errors
+
+
+def int_between(lower, upper):
+    def is_between(value):
+        if isinstance(value, int) and lower <= value <= upper:
+            return True
+        raise SchemaError(f"should be an int between {lower} and {upper}")
+
+    return is_between
+
+
+def float_between_with_halfstep(lower, upper):
+    def is_between(value):
+        is_number = isinstance(value, int) or isinstance(value, float)
+        is_half_step = re.match(r"^\d+(\.[05])?$", str(value))
+
+        if is_number and is_half_step and lower <= value <= upper:
+            return True
+        raise SchemaError(f"should be a number between {lower} and {upper} in half steps")
+
+    return is_between
+
+
+ENV_NAME = Regex(
+    r"^[a-zA-Z][a-zA-Z0-9]*$",
+    error="Environment name {} is invalid: names must only contain alphanumeric characters."
+    # It would be nice if this worked, but it doesn't seem to for dictionary keys.
```

Comment:
Is this comment still relevant? It looks like you've got it working downstairs.

Reply:
Yes, it is still relevant. For values the "error" parameter works and outputs the custom text. For keys the custom text doesn't get reported in the exception for some reason.

Reply:
```suggestion
    # For values the "error" parameter works and outputs the custom text. For keys the custom text doesn't get reported in the exception for some reason.
```